framework/src/modules/dpos/transaction_assets/pom_transaction_asset.ts

```diff
@@ -0,0 +1,264 @@
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { BaseBlockHeader } from '@liskhq/lisk-chain';
+import { getAddressFromPublicKey } from '@liskhq/lisk-cryptography';
+import { codec } from '@liskhq/lisk-codec';
+import { ApplyAssetInput, BaseAsset, ValidateAssetInput } from '../../base_asset';
+import { ValidationError } from '../../../errors';
+import { MAX_PUNISHABLE_BLOCK_HEIGHT_DIFFERENCE, MAX_POM_HEIGHTS } from '../constants';
+import { DPOSAccountProps, PomTransactionAssetInput } from '../types';
+import { getPunishmentPeriod, validateSignature } from '../utils';
+
+const signingBlockHeaderSchema = {
+	$id: 'lisk/signing-block-header',
+	type: 'object',
+	properties: {
+		version: { dataType: 'uint32', fieldNumber: 1 },
+		timestamp: { dataType: 'uint32', fieldNumber: 2 },
+		height: { dataType: 'uint32', fieldNumber: 3 },
+		previousBlockID: { dataType: 'bytes', fieldNumber: 4 },
+		transactionRoot: { dataType: 'bytes', fieldNumber: 5 },
+		generatorPublicKey: { dataType: 'bytes', fieldNumber: 6 },
+		reward: { dataType: 'uint64', fieldNumber: 7 },
+		asset: {
+			type: 'object',
+			fieldNumber: 8,
+			properties: {
+				maxHeightPreviouslyForged: {
+					dataType: 'uint32',
+					fieldNumber: 1,
+				},
+				maxHeightPrevoted: {
+					dataType: 'uint32',
+					fieldNumber: 2,
+				},
+				seedReveal: {
+					dataType: 'bytes',
+					fieldNumber: 3,
+				},
+			},
+			required: ['maxHeightPreviouslyForged', 'maxHeightPrevoted', 'seedReveal'],
+		},
+	},
+	required: [
+		'version',
+		'timestamp',
+		'height',
+		'previousBlockID',
+		'transactionRoot',
+		'generatorPublicKey',
+		'reward',
+		'asset',
+	],
+};
+
+export const blockHeaderSchema = {
+	...signingBlockHeaderSchema,
+	$id: 'lisk/block-header',
+	properties: {
+		...signingBlockHeaderSchema.properties,
+		signature: { dataType: 'bytes', fieldNumber: 9 },
+	},
+};
+
+const getBlockHeaderBytes = (header: BaseBlockHeader): Buffer =>
+	codec.encode(signingBlockHeaderSchema, header);
+
+export class PomTransactionAsset extends BaseAsset<PomTransactionAssetInput> {
+	public baseFee = BigInt(0);
+	public name = 'pom';
+	public type = 3;
+	public assetSchema = {
+		$id: 'lisk/dpos/pom',
+		type: 'object',
+		required: ['header1', 'header2'],
+		properties: {
+			header1: {
+				...blockHeaderSchema,
+				fieldNumber: 1,
+			},
+			header2: {
+				...blockHeaderSchema,
+				fieldNumber: 2,
+			},
+		},
+	};
+
+	// eslint-disable-next-line class-methods-use-this
+	public validateAsset({ asset }: ValidateAssetInput<PomTransactionAssetInput>): void {
+		if (!asset.header1.generatorPublicKey.equals(asset.header2.generatorPublicKey)) {
+			throw new ValidationError(
+				'GeneratorPublicKey of each BlockHeader should match.',
+				asset.header1.generatorPublicKey.toString('base64'),
+			);
+		}
+
+		if (
+			Buffer.compare(getBlockHeaderBytes(asset.header1), getBlockHeaderBytes(asset.header2)) === 0
```

Comment:
why not `equal` ?

Reply:
Not sure, moved the existing code. `equals` works here as well.